tools/runners/test262-harness.py

```diff
@@ -609,18 +610,18 @@ def execute(command):
         stderr = TempFile(prefix="test262-err-")
         try:
             logging.info("exec: %s", str(args))
-            process = subprocess.Popen(
+            with subprocess.Popen(
                 args,
                 shell=False,
                 stdout=stdout.file_desc,
                 stderr=stderr.file_desc
-            )
-            timer = threading.Timer(TEST262_CASE_TIMEOUT, process.kill)
-            timer.start()
-            code = process.wait()
-            timer.cancel()
-            out = stdout.read()
-            err = stderr.read()
+            ) as process:
+                timer = threading.Timer(TEST262_CASE_TIMEOUT, process.kill)
+                timer.start()
+                code = process.wait()
+                timer.cancel()
+                out = stdout.read()
+                err = stderr.read()
```

Comment:
I am a bit confused about this code here. process.wait() has a timeout since py3.3, so this seems to be overly complex now. the code below _might_ achieve the same effect but simpler. (disclaimer: the snippet has not been tested)

```py
with subprocess.Popen(...) as process:
    try:
        process.wait(timeout=TEST262_CASE_TIMEOUT)
    except TimeoutExpired:
        process.kill()
out = stdout.read()
err = stderr.read()
```